lib/utils.py

```diff
@@ -73,27 +73,15 @@ def validate_url(string):
     return bool(checker.scheme in ('http', 'https', 'rtsp', 'rtmp') and checker.netloc)
 
 
-def get_supervisor_api_response(delay):
-    def function():
-        balena_supervisor_address = os.getenv('BALENA_SUPERVISOR_ADDRESS')
-        balena_supervisor_api_key = os.getenv('BALENA_SUPERVISOR_API_KEY')
-        headers = {'Content-Type': 'application/json'}
-
-        return requests.get('{}/v1/device?apikey={}'.format(
-            balena_supervisor_address,
-            balena_supervisor_api_key
-        ), headers=headers)
+def get_supervisor_api_response():
+    balena_supervisor_address = os.getenv('BALENA_SUPERVISOR_ADDRESS')
+    balena_supervisor_api_key = os.getenv('BALENA_SUPERVISOR_API_KEY')
+    headers = {'Content-Type': 'application/json'}
 
-    return retry_call(
-        function,
-        delay=delay,
-        exceptions=(
-            requests.ConnectionError,
-            requests.ConnectTimeout,
-            requests.Timeout,
-            NewConnectionError,
-        ),
-    )
```

Comment:
@vpetersson, I temporarily removed the usage of the `retry` library to test a hypothesis.